### PR TITLE
Dt #165775587 migrate functions entities moves

### PR DESCRIPTION
**Pull request by donaldthai**
## Description

We're consolidating to the new paradigm for initiating swagger api requests, therefore, we have to refactor the old redux `Move` reducers. Selectors will also be affected and migrated.

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/165775587) for this change

### CODE REVIEW

File: src/scenes/Moves/MoveTypeWizard.jsx [hunk 1]
```diff
@@ -48,6 +50,11 @@ function mapDispatchToProps(dispatch) {
   return bindActionCreators({ updateMove, loadMove }, dispatch);
 }
 function mapStateToProps(state) {
-  return state.moves;
+  const move = getCurrentMove(state);
+
+  return {
+    currentMove: move,
+    pendingMoveType: get(state, 'moves.pendingMoveType'), //ToDo: will want to move away from using the move reducer later for this...
```
Comment (garrettqmartin8) [inline src/scenes/Moves/MoveTypeWizard.jsx]: @Igarfinkle is working on this 🎉

File: src/shared/utils.js [hunk 2]
```diff
@@ -36,6 +36,10 @@ export const upsert = (arr, newValue) => {
   }
 };
 
+export function fetchFirst(collection) {
+  return head(collection) || null;
```
Comment (garrettqmartin8) [inline src/shared/utils.js]: Is there a specific reason for using `head` instead of something like `collection[0]`?

File: src/shared/UI/ducks.js [hunk 3]
```diff
@@ -50,3 +68,11 @@ export function getCurrentShipmentID(state) {
 export function getCurrentShipment(state) {
   return selectShipment(state, get(state, 'ui.currentShipmentID'));
 }
+
+export function getCurrentMoveID(state) {
```
Comment (garrettqmartin8) [inline src/shared/UI/ducks.js]: I'm a fan of using the UI reducer for things that don't naturally fit inside the model-backed entities pattern.

File: src/shared/utils.js (hunk 2 above)
Comment (donaldthai) [reply src/shared/utils.js]: oh, i forgot to remove the `|| null`. `head()` returns the first element but defaults to null. :)

Comment (garrettqmartin8) [reply src/shared/utils.js]: Ah, gotcha.

File: src/shared/UI/ducks.js (hunk 3 above)
Comment (donaldthai) [reply src/shared/UI/ducks.js]: Should I refactor this out? Found it hard to keep track of the current move for the app

Comment (garrettqmartin8) [reply src/shared/UI/ducks.js]: Not sure where else I'd put it. I think it works pretty well in this context. These little persistent UI state things don't really fit into the new architecture, but I think it's ok for now.

File: src/scenes/Moves/ducks.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { get, head, pick } from 'lodash';
-import { UpdateMove, GetMove, SubmitMoveForApproval } from './api.js';
+import { SubmitMoveForApproval } from './api.js';
```
Comment (garrettqmartin8) [inline src/scenes/Moves/ducks.js]: Are we still using this from `api.js`?

Comment (donaldthai) [reply src/scenes/Moves/ducks.js]: Doesn't seem like it. I'll remove!

**Pull request merged**